### ISSUE TITLE
[TEST] Run e2e test cases on Photon OS to see if any failures there

### ISSUE THREAD

**Issue by yangchiu** [longhorn/longhorn#8113]
## What's the test to develop? Please describe

Run e2e test cases https://vmware.github.io/photon/ to see if any failures there.

**Comment by yangchiu** [comment]
I tried to [create a photon OS `AWS AMI`](https://docs.aws.amazon.com/vm-import/latest/userguide/required-permissions.html) to start a ec2 instance based on it, but it's not possible because of insufficient permission for `iam:CreateRole`: 
```
aws iam create-role --role-name vmimport --assume-role-policy-document trust-policy.json

An error occurred (AccessDenied) when calling the CreateRole operation: User: arn:aws:iam::641769369267:user/yang.chiu is not authorized to perform: iam:CreateRole on resource: arn:aws:iam::641769369267:role/vmimport because no identity-based policy allows the iam:CreateRole action
```

@innobead Is it possible for you to help add the permission? Thank you!

**Comment by innobead** [comment]
@yangchiu is it working now?

**Comment by yangchiu** [comment]
Test results based on [Photon OS 5.0 GA](https://github.com/vmware/photon/wiki/Downloading-Photon-OS#downloading-photon-os-50-ga):

[amd64](https://ci.longhorn.io/job/public/job/v1.4.x/job/v1.4.x-longhorn-tests-sles-amd64/473/): 8 failures
[arm64](https://ci.longhorn.io/job/public/job/v1.4.x/job/v1.4.x-longhorn-tests-sles-amd64/474/): 4 failures

Failed test cases are:
- `test_ha_backup_deletion_recovery`:
  - Flaky test case on master pipeline as well
- `test_engine_image_not_fully_deployed_perform_auto_upgrade_engine`:
  - Flaky test case on master pipeline as well
- `test_volume_metrics`:
  -  Known issue https://github.com/longhorn/longhorn/issues/7626#issuecomment-1984969057
- `test_engine_live_upgrade_while_replica_concurrent_rebuild`:
  - Timeout issue. Passed after increased the timeout value: https://ci.longhorn.io/job/public/job/v1.4.x/job/v1.4.x-longhorn-tests-sles-amd64/487/ 
- `test_setting_concurrent_rebuild_limit`:
  - Timeout issue. Passed after increased the timeout value: https://ci.longhorn.io/job/public/job/v1.4.x/job/v1.4.x-longhorn-tests-sles-amd64/484/ 
- `test_setting_concurrent_volume_backup_restore_limit`:
  - Timeout issue. Passed after increased the timeout value: https://ci.longhorn.io/job/public/job/v1.4.x/job/v1.4.x-longhorn-tests-sles-amd64/485/  
- `test_setting_concurrent_volume_backup_restore_limit_should_not_effect_dr_volumes`:
  - Timeout issue. Passed after increased the timeout value: https://ci.longhorn.io/job/public/job/v1.4.x/job/v1.4.x-longhorn-tests-sles-amd64/486/ 

In conclusion, there is no outstanding issue on `Photon OS 5.0` Longhorn regression tests.

cc @innobead

**Comment by innobead** [comment]
We need to check if the original test ran with `Realtime kernel parameter enabled`. If not, we need to rerun it if it's feasible on our test bed. 

@chriscchien Can you help with this? @yangchiu is OOO this week.

**Comment by chriscchien** [comment]
From the [GA](https://github.com/vmware/photon/wiki/Downloading-Photon-OS#downloading-photon-os-50-ga) page, if I understand correctly only [ISO x86_64 Real-Time flavour](https://packages.vmware.com/photon/5.0/GA/iso/photon-rt-5.0-dde71ec57.x86_64.iso) support `Realtime kernel parameter enabled` (I tried use `tdnf install linux-rt` to update kernel in ec2 instance, but instance will not ready after reboot). 

Currently I have some issue while setting up cluster with ISO file. I will continue work on this and may need more time, thank you.

**Comment by mantissahz** [comment]
Update for the regression test in my local environment on Photon5 RT OS:
1. test_support_bundle.py::test_support_bundle_should_not_timeout:
   Need 11G Memory at least for worker nodes.
2. test_settings.py::test_instance_manager_cpu_reservation:
   Need fixed 4 CPUs worker nodes.
Other test cases are almost mentioned as https://github.com/longhorn/longhorn/issues/8113#issuecomment-1985371533

**Comment by innobead** [comment]
> Update for the regression test in my local environment on Photon5 RT OS:
> 
> 1. test_support_bundle.py::test_support_bundle_should_not_timeout:
>    Need 11G Memory at least for worker nodes.
> 2. test_settings.py::test_instance_manager_cpu_reservation:
>    Need fixed 4 CPUs worker nodes.
>    Other test cases are almost mentioned as [[TEST] Run e2e test cases on Photon OS to see if any failures there #8113 (comment)](https://github.com/longhorn/longhorn/issues/8113#issuecomment-1985371533)

@mantissahz these test cases are intended. Did we encounter any issues with other cases?

**Comment by mantissahz** [comment]
>these test cases are intended. Did we encounter any issues with other cases?

No, no other cases are outstanding.

**Comment by mantissahz** [comment]
Ref: https://github.com/longhorn/longhorn/issues/8253

**Comment by yangchiu** [comment]
I've tried different approaches but still unable to create a valid AMI to run an EC2 instances. Need more time to investigate.

**Comment by innobead** [comment]
No worries. No need to spend extra time on this image build (rt). We already spent enough time on this. Revisit this if needed.
